### ISSUE TITLE
Gamepad buttons are always bound to multiple functions

### ISSUE THREAD

**Issue by second2050** [FWGS/xash3d-fwgs#1337]
# Issue
Gamepad buttons always execute to functions: one function I defined and one function that I haven't defined but that is always the same for a given button.
This happens with both a DualShock 4 v2 and a Steam Controller (used via SC Controller, emulating an Xbox 360 Controller).

## Example
 - A / Cross button: Defined as `+jump` but also always executes `+attack2`
 - B / Circle button: Defined as `+reload` but also always executes `+attack`
 - LB / L1: Defined as `invprev` but also always executes `+duck`
 - RS / R3: Defined as `+use` but also always executes `pause`
 
 ## Things I have tried
 - using `unbindall` to remove all bindings and then rebind all the buttons I need
 - executing xash3d with the `-sdl_joy_old_api` flag, which makes it not recognize controllers anymore

# Platform information
OS: Arch Linux (Kernel `6.3.0-x64v3-xanmod1-1`)
Xash3D version: `Xash3D FWGS 49/0.20 (linux-i386 build 2947)` compiled from master (48176233bdc264cd4e5bce1f8df83f2f4bc59f7c)

## Logs and attachments
[engine.log](https://github.com/FWGS/xash3d-fwgs/files/11331831/engine.log)
[xgamepad.cfg.txt](https://github.com/FWGS/xash3d-fwgs/files/11331845/xgamepad.cfg.txt) Gamepad Bindings

**Comment by mmillar-bolis** [comment]
Are both the DualShock 4 and the Steam Controller being passed through SC-Controller? In your log, on line 116, it shows `joystick found X360 Controller`, so I wasn't sure which device it was an example of. However, if you are using both of these devices through the virtual driver, especially over bluetooth, here is an [issue very similar](https://github.com/ValveSoftware/Proton/issues/4707#issuecomment-812681212) to the one you are encountering.

As of SDL 2.0.12, there is a new addition, `HIDAPI`, which can read HID devices that are in raw mode. This is especially helpful in supporting the gyroscopic functionality found in more recent controllers. However, if a device driver can expose hidraw devices as user-readable, due to the way that SDL2 works, it can basically sidestep some virtual drivers to talk to an hidraw device more directly. Because SDL2 is getting feedback from effectively two sources, it may make some wrong assumptions about the protocol mode of a device and improperly pick up on competing hardware button assignments.

This double-input behavior might be solved by setting the environment variable `SDL_JOYSTICK_HIDAPI=0` while running Xash3D. It also may be worth checking if the problem persists when using the kernel's own DS4 driver instead. Likewise, if the Steam userspace driver is running, it has been known to cause input conflicts with SC-Controller.

---

To explain issues with the `-sdl_joy_old_api` arg: as of Xash3D-FWGS 0.19.1, there are two controller implementations, a modern one that is functionally similar to most current Quake sourceports, and a legacy one that still functions in the same manner as GoldSrc. The newer "gamepad" functionality leverages the abstracted `SDL_GameController` API while the older "joystick" functionality leverages the more raw `SDL_Joystick` API. Turning on the legacy joystick support turns off the modern gamepad abstraction. Because of this, when using the old Joystick API, you need to set up your controller independently by using the console commands that are related to it's functions, namely:

```
// Enable or disable
joystick 1
// Configure axis (only 1st joystick is supported in GoldSrc)
joyadvanced 1
joyadvaxisx 3
joyadvaxisy 1
joyadvaxisz 0
joyadvaxisr 0
joyadvaxisu 0
joyadvaxisv 0
// For analogue movement
joyforwardsensitivity -1
joyforwardthreshold  0.15
joysidesensitivity 1
joysidethreshold 0.15
// Apply settings
joyadvancedupdate
//
// Face buttons are usually JOY1~JOY4 while all others
// are AUX1~AUX16 (varies by controller). In GoldSrc,
// analogue triggers and secondary joystick are not
// generally supported.
```

If you didn't go through this secondary setup process for the old API, it could explain why your controllers appeared to no longer function when switching to it.

**Comment by a1batross** [comment]
Better set `joystick 0` in this config.

It seems that joystick support in original client.dll is conflicting with our in-engine support. You have to disable one of them.

I will figure out how to support both later.

**Comment by second2050** [comment]
According to the console `joystick 0` is already set and executing `joystick 0` doesn't seem to change anything.
Setting `joystick 1` makes the left stick also control the camera though.

**Comment by CkauNui** [comment]
[RU] Такая же проблема, всё ещё нет исправленного client.dll?

**Comment by OlegAckbar** [comment]
I can confirm this issue. For me, A and B buttons are also acting as primary and secondary attacks, even though I've not set them so.

[xash3d_gamepad.webm](https://github.com/user-attachments/assets/0d8acc64-8837-4f60-89a1-26a9bee6f792)

**Comment by OlegAckbar** [comment]
hlsdk-portable libraries are fixing the issue

**Comment by a1batross** [comment]
Yeah, hlsdk-portable client wouldn't have this issue because it's aware of our custom input interface.
